### PR TITLE
[RFR] Remove local VM fixtures and replace usage with global create_vm fixture.

### PR DESCRIPTION
**Pull request by prichard77**
## Purpose or Intent

- __Updating tests__ to use a common create_vm fixture instead of local ones.

### PRT Run

{{ pytest: --long-running cfme/tests/infrastructure/test_vm_migrate.py::test_vm_migrate cfme/tests/infrastructure/test_vm_ownership.py::test_rename_vm --long-running cfme/tests/infrastructure/test_vm_power_control.py cfme/tests/infrastructure/test_vm_reconfigure.py }}

### CODE REVIEW

**Comment by prichard77** [comment]
test_vm_migrate did previously use a fixture that put "migrate" into the vm name. I don't believe we have support for altering the name in create_vm but I'll double check. Should we even worry about adding "migrate' to the name?

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 921f61212f8bd47c2e2646276dd7e1205092cab1

The *local* fixture **`vm_name`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**
    - *testing_vm_tools*

The *local* fixture **`archived_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**
    - *test_no_power_controls_on_archived_vm*
    - *test_archived_vm_status*

The *local* fixture **`orphaned_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**
    - *test_orphaned_vm_status*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit c068c7a3ed9b299646ed49666dce35c8685adc31

The *local* fixture **`vm_name`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**

The *local* fixture **`archived_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**
    - *test_no_power_controls_on_archived_vm*
    - *test_archived_vm_status*

The *local* fixture **`orphaned_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**
    - *test_orphaned_vm_status*

The *local* fixture **`archive_orphan_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**
    - *test_power_options_on_archived_orphaned_vms_all_page*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 49d239e731eb409ab5590181643009c341397712

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *local* fixture **`vm_name`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**

The *local* fixture **`archived_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**
    - *test_no_power_controls_on_archived_vm*
    - *test_archived_vm_status*

The *local* fixture **`orphaned_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**
    - *test_orphaned_vm_status*

The *local* fixture **`archive_orphan_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**
    - *test_power_options_on_archived_orphaned_vms_all_page*

The *local* fixture **`ensure_vm_stopped`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_reconfigure.py**
    - *enable_hot_plugin*
    - *test_vm_reconfig_add_remove_hw_cold*
    - *test_reconfig_vm_negative_cancel*

The *local* fixture **`ensure_vm_running`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_reconfigure.py**

The *local* fixture **`vm_state`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_reconfigure.py**
    - *test_vm_reconfig_add_remove_disk*
    - *test_vm_reconfig_resize_disk*

The *local* fixture **`enable_hot_plugin`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_reconfigure.py**

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit e5e2f1737a17df82f61e8911adfd7f3ca30f6da0

The *local* fixture **`vm_name`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**

The *local* fixture **`archived_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**
    - *test_no_power_controls_on_archived_vm*
    - *test_archived_vm_status*

The *local* fixture **`orphaned_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**
    - *test_orphaned_vm_status*

The *local* fixture **`archive_orphan_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**
    - *test_power_options_on_archived_orphaned_vms_all_page*

The *local* fixture **`ensure_vm_stopped`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_reconfigure.py**
    - *enable_hot_plugin*
    - *test_vm_reconfig_add_remove_hw_cold*
    - *test_reconfig_vm_negative_cancel*

The *local* fixture **`ensure_vm_running`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_reconfigure.py**

The *local* fixture **`enable_hot_plugin`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_reconfigure.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by prichard77** [comment]
Some of these are failing and I am going thru and trying to determine if the tests were failing initially. 
test_vm_migrate is failing for rhv4.3 but it fails with initial code. I don't own these tests, so I am not familiar with the history. Looking into now.

**Comment by prichard77** [comment]
making RFR so I can see in active PRs on Ostriz

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 9e7bb41ca57fbe1d10b1de6faf5c46ef5bdb72c3

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *local* fixture **`vm_name`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**

The *local* fixture **`archived_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**
    - *test_no_power_controls_on_archived_vm*
    - *test_archived_vm_status*

The *local* fixture **`orphaned_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**
    - *test_orphaned_vm_status*

The *local* fixture **`archive_orphan_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**
    - *test_power_options_on_archived_orphaned_vms_all_page*

The *local* fixture **`ensure_vm_stopped`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_reconfigure.py**
    - *enable_hot_plugin*
    - *test_vm_reconfig_add_remove_hw_cold*
    - *test_reconfig_vm_negative_cancel*

The *local* fixture **`ensure_vm_running`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_reconfigure.py**

The *local* fixture **`vm_state`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_reconfigure.py**
    - *test_vm_reconfig_add_remove_disk*
    - *test_vm_reconfig_resize_disk*

The *local* fixture **`enable_hot_plugin`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_reconfigure.py**

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by prichard77** [comment]
test_vm_ownership.py - test_rename_vm[virtualcenter] fails in cleanup. Is this related to the fact we did rename?

**Comment by prichard77** [comment]
Here is the rest of my notes from retesting the last PRT run. I am including the ones I put above so they will be in one place.

test_vm_migrate.py - test_vm_migrate[rhevm-4.3] retesting with 4.3 - Failed. Still fails when I add old code back in. 
                   - test_vm_migrate[rhevm-4.4] is passing in PRT.

test_vm_ownership.py - test_rename_vm[virtualcenter] fails in cleanup. Is this related to the fact we did rename?
                
test_vm_power_control.py - TestVmDetailsPowerControlPerProvider.test_power_off[virtualcenter-6.5] - passes locally
			   TestVmDetailsPowerControlPerProvider.test_power_on[rhevm-4.3] - passes locally
			   TestVmDetailsPowerControlPerProvider.test_power_on[rhevm-4.4] - failed locally first try, passed second
                           test_guest_os_shutdown[virtualcenter-6.5-full_template] timed out in PRT. Passed locally on retest.	   
		These all failed because of: Exception: Provider collection empty
			   These are orphaned VMs. Which means they have been disassociated with the provider. So wouldn't we expect the 
                              provider collection to be empty. We might need to add an "orphaned" option to create_view to support this?
		           test_orphaned_vm_status[rhevm-4.3] - passed on local retest? but failed teardown? 
                           test_orphaned_vm_status[rhevm-4.4] - passed on local retest? but failed teardown? 
                           test_orphaned_vm_status[virtualcenter-6.5] - passed on local retest? but failed teardown? 
                           test_power_options_on_archived_ophaned_vms_all_page[orphaned-rhevm-4.3]
			   test_power_options_on_archived_ophaned_vms_all_page[orphaned-rhevm-4.4]
                           test_power_options_on_archived_ophaned_vms_all_page[virtualcenter-6.5]

**Comment by prichard77** [comment]
i believe the orphan tests are failing teardown because they are orphans. We are getting failures for an empty Provider collection and I believe this would be expected for orphans. Still looking for confirmation.

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 405ed892b6417d73e391c7747f5d1ad1e015c7b7

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *local* fixture **`vm_name`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**

The *local* fixture **`archived_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**
    - *test_no_power_controls_on_archived_vm*
    - *test_archived_vm_status*

The *local* fixture **`orphaned_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**
    - *test_orphaned_vm_status*

The *local* fixture **`archive_orphan_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**
    - *test_power_options_on_archived_orphaned_vms_all_page*

The *local* fixture **`ensure_vm_stopped`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_reconfigure.py**
    - *enable_hot_plugin*
    - *test_vm_reconfig_add_remove_hw_cold*
    - *test_reconfig_vm_negative_cancel*

The *local* fixture **`ensure_vm_running`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_reconfigure.py**

The *local* fixture **`vm_state`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_reconfigure.py**
    - *test_vm_reconfig_add_remove_disk*
    - *test_vm_reconfig_resize_disk*

The *local* fixture **`enable_hot_plugin`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_reconfigure.py**

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by prichard77** [comment]
I updated _cleaup in _create_vm to catch the "Provider collection empty" exception.

**Comment by prichard77** [comment]
I do believe `test_vm_ownership.py::test_rename_vm[virtualcenter]` is failing in cleanup because we renamed the vm. I need to locate the actual cause and see if it a bug or something we need to update to support. Or maybe update the testcase to rename the vm back to the original.

File: cfme/fixtures/vm.py [hunk 1]
```diff
@@ -49,7 +49,11 @@ def _create_vm(request, template_type, provider, vm_name):
     @request.addfinalizer
     def _cleanup():
         vm_obj.cleanup_on_provider()
-        provider.refresh_provider_relationships()
+        try:
+            provider.refresh_provider_relationships()
+        except Exception as e:
```
Comment (jaryn) [inline cfme/fixtures/vm.py:54]: Could you please be more specific about which Exception can it be?

Comment (prichard77) [reply cfme/fixtures/vm.py:54]: Sure. And then also catch the more specific LookupError instead of just an Exception in _cleanup().

Comment (prichard77) [reply cfme/fixtures/vm.py:54]: I don't know why we'd want the else clause, however.

Comment (prichard77) [reply cfme/fixtures/vm.py:54]: Changing exception but not adding else.

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 7b27c299230d3b7a8f7dbbeba4f50038b4de8e59

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *local* fixture **`vm_name`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**

The *local* fixture **`archived_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**
    - *test_no_power_controls_on_archived_vm*
    - *test_archived_vm_status*

The *local* fixture **`orphaned_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**
    - *test_orphaned_vm_status*

The *local* fixture **`archive_orphan_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**
    - *test_power_options_on_archived_orphaned_vms_all_page*

The *local* fixture **`ensure_vm_stopped`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_reconfigure.py**
    - *enable_hot_plugin*
    - *test_vm_reconfig_add_remove_hw_cold*
    - *test_reconfig_vm_negative_cancel*

The *local* fixture **`ensure_vm_running`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_reconfigure.py**

The *local* fixture **`vm_state`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_reconfigure.py**
    - *test_vm_reconfig_add_remove_disk*
    - *test_vm_reconfig_resize_disk*

The *local* fixture **`enable_hot_plugin`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_reconfigure.py**

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by prichard77** [comment]
I updated cleanup_on_provider to catch TimeOutError and log it. This fixes the failure on cleanup for test_rename_vm.

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 08ac10389bfa2e5ae0976acbff3a8a9e7ed0e39c

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *local* fixture **`vm_name`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**

The *local* fixture **`archived_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**
    - *test_no_power_controls_on_archived_vm*
    - *test_archived_vm_status*

The *local* fixture **`orphaned_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**
    - *test_orphaned_vm_status*

The *local* fixture **`archive_orphan_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_power_control.py**
    - *test_power_options_on_archived_orphaned_vms_all_page*

The *local* fixture **`ensure_vm_stopped`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_reconfigure.py**
    - *enable_hot_plugin*
    - *test_vm_reconfig_add_remove_hw_cold*
    - *test_reconfig_vm_negative_cancel*

The *local* fixture **`ensure_vm_running`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_reconfigure.py**

The *local* fixture **`vm_state`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_reconfigure.py**
    - *test_vm_reconfig_add_remove_disk*
    - *test_vm_reconfig_resize_disk*

The *local* fixture **`enable_hot_plugin`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_reconfigure.py**

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Pull request merged**